pkg/oauth/tokenrequest/basicauth_test.go

```diff
@@ -0,0 +1,347 @@
+					}
+					if out.String() != challenge.ExpectedPrompt {
+						t.Errorf("%s: %d: Expected prompt %q, got %q", k, i, challenge.ExpectedPrompt, out.String())
+					}
+				}
+			})
+		}
+	}
+}
+
+func TestBasicRealm(t *testing.T) {
+
+	testCases := map[string]struct {
+		Headers       http.Header
+		ExpectedBasic bool
+		ExpectedRealm string
+	}{
+		"empty": {
+			Headers:       http.Header{},
+			ExpectedBasic: false,
+			ExpectedRealm: ``,
+		},
+
+		"non-challenge": {
+			Headers: http.Header{
+				"test": []string{`value`},
+			},
+			ExpectedBasic: false,
+			ExpectedRealm: ``,
+		},
+
+		"non-basic": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{
+					`basicrealm="myrealm"`,
+					`digest basic="realm"`,
+				},
+			},
+			ExpectedBasic: false,
+			ExpectedRealm: ``,
+		},
+
+		"basic multiple www-authenticate headers": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{
+					`digest realm="digestrealm"`,
+					`basic realm="Foo"`,
+					`foo bar="baz"`,
+				},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo`,
+		},
+
+		"basic no realm": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: ``,
+		},
+
+		"basic other param": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic otherparam="othervalue"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: ``,
+		},
+
+		"basic token realm": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic realm=Foo Bar `},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo Bar`,
+		},
+
+		"basic quoted realm": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic realm="Foo Bar"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo Bar`,
+		},
+
+		"basic case-insensitive scheme": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`BASIC realm="Foo"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo`,
+		},
+
+		"basic case-insensitive realm": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic REALM="Foo"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo`,
+		},
+
+		"basic whitespace": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{` 	basic 	realm 	= 	"Foo\" Bar" 	`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo\" Bar`,
+		},
+
+		"basic trailing comma": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic realm="Foo", otherparam="value"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo`,
+		},
+
+		"realm containing quotes": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic realm="F\"oo", otherparam="value"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `F\"oo`,
+		},
+
+		"realm containing comma": {
+			Headers: http.Header{
+				WWW_AUTHENTICATE: []string{`basic realm="Foo, bar", otherparam="value"`},
+			},
+			ExpectedBasic: true,
+			ExpectedRealm: `Foo, bar`,
+		},
+
+		// TODO: additional forms to support
```

Comment:
wouldn't this be a good timing to add these?

Reply:
this does not seem to be about the tests but about the general support of these forms, I don't think we should extend the functionality as a part of this PR